src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
+        notifyProgress(source, PROGRESS_MAX);
+        try (InputStream in = selectDecompressor(source, new FileInputStream(source));
+             JsonParser jp = new JsonFactory().createParser(in)) {
+            readData(jp, listeFilme);
         } catch (FileNotFoundException ex) {
             Log.errorLog(894512369, "FilmListe existiert nicht: " + source);
             listeFilme.clear();
         } catch (Exception ex) {
             Log.errorLog(945123641, ex, "FilmListe: " + source);
             listeFilme.clear();
         }
+    }
+
+    private void checkDays(long days) {
+        if (days > 0) {
+            milliseconds = System.currentTimeMillis() - TimeUnit.MILLISECONDS.convert(days, TimeUnit.DAYS);
+        } else {
+            milliseconds = 0;
+        }
+    }
 
-        if (Config.getStop()) {
-            Log.sysLog("--> Abbruch");
+    public void readFilmListe(String source, final ListeFilme listeFilme, int days) {
+        try {
+            Log.sysLog("Liste Filme lesen von: " + source);
             listeFilme.clear();
+            this.notifyStart(source, PROGRESS_MAX); // für die Progressanzeige
+
+            checkDays(days);
+
+            if (!source.startsWith("http")) {
+                processFromFile(source, listeFilme);
+            } else {
+                processFromWeb(new URL(source), listeFilme);
+            }
+
+            if (Config.getStop()) {
+                Log.sysLog("--> Abbruch");
+                listeFilme.clear();
+            }
+        } catch (MalformedURLException ex) {
+            ex.printStackTrace();
         }
+
         notifyFertig(source, listeFilme);
     }
 
+    /**
+     * Download a process a filmliste from the web.
+     *
+     * @param source     source url as string
+     * @param listeFilme the list to read to
+     */
+    private void processFromWeb(URL source, ListeFilme listeFilme) {
+        Request.Builder builder = new Request.Builder().url(source);
+        builder.addHeader("User-Agent", Config.getUserAgent());
+
+        //our progress monitor callback
+        InputStreamProgressMonitor monitor = new InputStreamProgressMonitor() {
+            private int oldProgress = 0;
+
+            @Override
+            public void progress(long bytesRead, long size) {
+                final int iProgress = (int) (bytesRead * 100 / size);
+                if (iProgress != oldProgress) {
+                    oldProgress = iProgress;
+                    notifyProgress(source.toString(), iProgress);
+                }
+            }
+        };
+
+        try (Response response = MVHttpClient.getInstance().getHttpClient().newCall(builder.build()).execute();
+             ResponseBody body = response.body()) {
+            if (response.isSuccessful()) {
+                try (InputStream input = new ProgressMonitorInputStream(body.byteStream(), body.contentLength(), monitor)) {
+                    try (InputStream is = selectDecompressor(source.toString(), input);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)